slurmutils/models/slurm.py

```diff
@@ -310,6 +313,23 @@ def partitions(self):
         """Delete entire partition mapping in the Slurm configuration."""
         self.data["Partitions"] = {}
 
+    def _update_dict_property(self, other: Dict[str, Any], prop: str):
+        for k, v in other[prop].items():
+            item = self.data[prop].get(k, {})
+            item.update(v)
+            self.data[prop][k] = item
+
+        del other[prop]
+
+    def override(self, other: "SlurmConfig"):
+        """Override the fields of this model with the fields of another model."""
+        other = other.dict()
+
+        for key in ["Nodes", "FrontendNodes", "NodeSets", "Partitions"]:
+            self._update_dict_property(other, key)
+
+        self.data.update({k: v for k, v in other.items() if v})
```

Comment:
Also, there's no special handling for `DownNodes`. Is this intentional? Reason saying is that `DownNodes` in the internal dict representation is seeded as an empty list if no `DownNodes` configuration is supplied when constructing a new `SlurmConfig` object:

https://github.com/charmed-hpc/slurmutils/blob/6621623b3f77126c45f3cfcb7fc25835e13d15b8/slurmutils/models/slurm.py#L233

This means that this empty list will replace any DownNodes configuration in the current configuration, which I'm not sure is the behavior that we want? Perhaps we take the intersection of current and other's DownNodes list configuration, and then append any additional DownNodes defined in other? Let me know your thoughts here.

Reply:
Yeah, that sounds fair. I'll inline that function.

> Also, there's no special handling for DownNodes.

The `k: v for k, v in other.items() if v` skips `DownNodes` if it's empty. I agree that it would be better to just append instead of overriding in this case, but I don't think we can take the intersection; `DownNodes` doesn't have an "obvious" primary key, it's just an array of node names and some properties, and intersecting based off of its list of down nodes sounds a bit wonky.

Reply:
We'd probably need some form of recursive sorting function, but that's out of scope for this PR :mag_right: